opal/corelib/runtime.js

```diff
@@ -1817,9 +1817,16 @@
     }
 
     new_body = function() {
-      body.$$p = new_body.$$p;
-      new_body.$$p = null;
-      return body.apply(this, arguments);
+      var block = new_body.$$p;
+      var args = new Array(arguments.length);
+
+      for(var index = 0; index < arguments.length; index++) {
```

Comment:
Sure, that makes sense. This PR was mostly to throw ideas at the wall to see if they stick. :-)

I think I checked the performance tradeoff between allocating another local variable in the stack and calling `arguments.length` a long time ago for a PR. I remember the local variable was sometimes slightly faster and sometimes break-even depending on the browser, but never slower, so it's probably a good idea especially for a method that is likely to be called within a loop.